bin/wasm-node/rust/src/cpu_rate_limiter.rs

```diff
@@ -44,11 +44,17 @@ impl<T> CpuRateLimiter<T> {
     /// `u32::max_value()` represents "one CPU". For example passing `rate_limit / 2` represents
     /// "50% of one CPU".
     pub fn new(inner: T, rate_limit: u32) -> Self {
+        let max_divided_by_rate_limit_minus_one =
+            (f64::from(u32::max_value()) / f64::from(rate_limit)) - 1.0;
+        // Note that it is legal for `max_divided_by_rate_limit_minus_one` to be +infinite
+        debug_assert!(
```

Comment:
is there a guidance on when to use debug_asserts? don't you think they tend to clutter the code?

Reply:
I don't think I've wrote this down, but I use them for internal-bug-detecting purposes.
In other words, what you put in the `debug_assert!` is something that, if false, indicates that there's a bug in the internal logic of the current module.
However if it's an incorrect API usage, then it should be `assert!` instead.